File: c/isogram/src/isogram.c
#include <ctype.h>
#include <stdlib.h>
#include <string.h>

#include "isogram.h"

bool is_isogram(const char phrase[])
{
        if (phrase == NULL)
                return false;

        int length = strlen(phrase);

        if (length < 2)
                return true;

        char ctable[25] = { 0 };
        char current = 0;
        int pos = 0;
    
        for (int i = 0; i < length; i++) {
                current = tolower(phrase[i]);

                if (current > 96 && current < 123) {
                        pos = current - 97;

                        if (ctable[pos] != current)
                                ctable[pos] = current;
                        else
                                return false;
                }
        }

        return true;
}

File: python/isogram/isogram.py
def is_isogram(string):
    """Determine if a word or phrase is an isogram.

    An isogram (also known as a "nonpattern word") is a word or phrase
    without a repeating letter, however spaces, hyphens and apostrophes
    are allowed to appear multiple times.

    Args:
        string: Word or phrase to be checked.

    Returns:
        True if string is an isogram, False otherwise.
    """
    if type(string) is not str:
        return False

    letters = string.replace(" ", "").replace("-", "").replace("'", "").lower()

    return len(letters) == len(set(letters))
